File: 0735-asteroid-collision/0735-asteroid-collision.js
var asteroidCollision = function(asteroids) {
    const arr = [];

    asteroids.forEach(ele =>{
        let isDestroyed = false;

        while(arr.length != 0 
            && arr[arr.length - 1] > 0 
            && ele < 0
            ){

            if(arr[arr.length - 1] < -ele){
                arr.pop();
            }else if(arr[arr.length - 1] == -ele){
                arr.pop();
                isDestroyed = true;
                break;
            }else{
                isDestroyed = true;
                break;
            }
        }
        if(!isDestroyed){
            arr.push(ele);
        }
    });

    return arr;
};

File: 0735-asteroid-collision/0735-asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> st = new Stack<>();

        int n = asteroids.length;

        for(int i = 0; i < n; i++){
            int val = asteroids[i];
            boolean isDestroyed = false;

            while(!st.isEmpty() && st.peek() > 0 && val < 0){

                if(st.peek() < -val){
                    st.pop();
                }else if(st.peek() == -val){
                    isDestroyed = true;
                    st.pop();
                    break;
                }else{
                    isDestroyed = true;
                    break;
                }
            }
            if(!isDestroyed){
                st.push(val);
            }
            
        }

        n  = st.size();

        int[] res = new int[n];
        int j = n - 1;

        while(!st.isEmpty()){
            res[j--] = st.pop();
        }

        return res;
    }
}